File: tests-cypress/cypress/support/keywords/dashboard.keywords.ts
import { dashboardPage } from '../pages/dashboard/dashboard-page'

export function action_openDashboard() {
  cy.log('Open the dashboard')

  dashboardPage.visit()
}

export function check_dashboardShellReady() {
  cy.log('Verify the dashboard stat cards, price chart and top movers render')

  dashboardPage.verifyStatsVisible()
  dashboardPage.verifyChartVisible()
  dashboardPage.verifyTopMoversVisible()
}

export function check_portfolioStats(transactionCount: number, watchlistCount: number) {
  cy.log(`Verify portfolio stats - transactions: ${transactionCount}, watchlist: ${watchlistCount}`)

  dashboardPage.verifyTransactionsCount(transactionCount)
  dashboardPage.verifyWatchlistCount(watchlistCount)
}

File: tests-selenium/src/test/java/com/coinpulse/selenium/keywords/DashboardKeywords.java
package com.coinpulse.selenium.keywords;

import com.coinpulse.selenium.pages.DashboardPage;
import io.qameta.allure.Allure;
import org.openqa.selenium.WebDriver;

/**
 * Dashboard keywords: specs call these, never the page object directly.
 */
public class DashboardKeywords {

    private final DashboardPage dashboardPage;

    public DashboardKeywords(WebDriver driver) {
        this.dashboardPage = new DashboardPage(driver);
    }

    public void actionOpenDashboard() {
        Allure.step("Open the dashboard", dashboardPage::open);
    }

    public void checkPortfolioStats(int transactionCount, int watchlistCount) {
        Allure.step("Verify portfolio stats - transactions: " + transactionCount
                + ", watchlist: " + watchlistCount, () -> {
            dashboardPage.verifyTransactionsCount(transactionCount);
            dashboardPage.verifyWatchlistCount(watchlistCount);
        });
    }

    public void checkDashboardShellReady() {
        Allure.step("Verify the dashboard stat cards, price chart and top movers render", () -> {
            dashboardPage.verifyStatsVisible();
            dashboardPage.verifyChartVisible();
            dashboardPage.verifyTopMoversVisible();
        });
    }
}
